torchgeo/datasets/oscd.py

```diff
@@ -414,22 +422,33 @@ def setup(self, stage: Optional[str] = None) -> None:
 
         This method is called once per GPU per run.
         """
-        transforms = Compose([self.preprocess])
+
+        def random_crop(sample: Dict[str, Any]) -> Dict[str, Any]:
+            sample["mask"] = repeat(sample["mask"], "h w -> t h w", t=2)
+            sample["image"], sample["mask"] = self.rcrop(
+                sample["image"], sample["mask"]
+            )
+            sample["mask"] = sample["mask"].squeeze()[0]  # type: ignore[attr-defined]
+            return sample
+
+        train_transforms = Compose([self.preprocess, random_crop])
+        test_transforms = Compose([self.preprocess])
 
         dataset = OSCD(
-            self.root_dir, split="train", bands=self.bands, transforms=transforms
+            self.root_dir, split="train", bands=self.bands, transforms=train_transforms
         )
 
         if self.val_split_pct > 0.0:
             self.train_dataset, self.val_dataset, _ = dataset_split(
                 dataset, val_pct=self.val_split_pct, test_pct=0.0
             )
+            self.val_dataset.transforms = test_transforms
```

Comment:
Unsure why this isn't working. The val_dataset and train_dataset are split from the same dataset and I need to override the val_datasets transform somehow.

Reply:
If you look at the return value of dataset_split you will see that it returns a torch.utils.data.Subset object. If you look up what a Subset object is you will find that it has an attribute Subset.dataset which holds the original dataset and samples indices from it. You can override the transforms via self.val_dataset.dataset.transforms